pkg/git/repository/github/service.go

```diff
@@ -56,24 +60,29 @@ func newGhService(gitSource *v1alpha1.GitSource, endpoint *gittransport.Endpoint
 		gitSource: gitSource,
 		client:    client,
 		repo:      repo,
+		secret:    secret,
 	}, nil
 }
 
-func (s *RepositoryService) GetListOfFilesInRootDir() ([]string, error) {
+func (s *RepositoryService) FileExistenceChecker() (repository.FileExistenceChecker, error) {
 	tree, _, err := s.client.Git.GetTree(
```

Comment:
Let's not do this at all for Github. Since we are anyway 'customizing' the implementation for Github, let's only use the `HEAD` requests if s.secret uses `anonymous`

Reply:
ok, I thought that we wouldn't miss anything in case we give it a try with the API call and fall back to `HEAD` requests just when we hit the rate limit. The are two limitations of the `HEAD` request approach:
* we cannot use regexp for matching the files (but that shouldn't be a big deal - only a few of them are really regexp)
* we cannot get the list of languages (this shouldn't be a problem if we detect the build tool, right?)

Looking at the pros and cons - yeah, it's probably better to not use the unauthenticated API calls and avoid hitting api rate limit unnecessarily.

Reply:
done 4f81465

Reply:
Yes I did wonder about language detection , but since we were anyway detecting the type, we are OK.

Reply:
Yes, I guess that language detection is kind of helper information in the cases when
* we wouldn't detect any build type, then we could assume it from the most used language
* we would find more build types, so the order of languages would prioritize them